src/main/java/bio/terra/workspace/service/resource/controlled/mappings/CustomGcpIamRoleMapping.java

```diff
@@ -0,0 +1,140 @@
+  // this map for completeness.
+  public static final Table<WsmResourceType, ControlledResourceIamRole, CustomGcpIamRole>
+      CUSTOM_GCP_IAM_ROLES =
+          new ImmutableTable.Builder<WsmResourceType, ControlledResourceIamRole, CustomGcpIamRole>()
+              // GCS bucket
+              .put(
+                  WsmResourceType.GCS_BUCKET,
+                  ControlledResourceIamRole.READER,
+                  new CustomGcpIamRole(
+                      WsmResourceType.GCS_BUCKET,
+                      ControlledResourceIamRole.READER,
+                      GCS_BUCKET_READER_PERMISSIONS))
+              .put(
+                  WsmResourceType.GCS_BUCKET,
+                  ControlledResourceIamRole.WRITER,
+                  new CustomGcpIamRole(
+                      WsmResourceType.GCS_BUCKET,
+                      ControlledResourceIamRole.WRITER,
+                      GCS_BUCKET_WRITER_PERMISSIONS))
+              .put(
+                  WsmResourceType.GCS_BUCKET,
+                  ControlledResourceIamRole.EDITOR,
+                  new CustomGcpIamRole(
+                      WsmResourceType.GCS_BUCKET,
+                      ControlledResourceIamRole.EDITOR,
+                      GCS_BUCKET_EDITOR_PERMISSIONS))
+              .put(
+                  WsmResourceType.GCS_BUCKET,
+                  ControlledResourceIamRole.ASSIGNER,
+                  new CustomGcpIamRole(
+                      WsmResourceType.GCS_BUCKET,
+                      ControlledResourceIamRole.ASSIGNER,
+                      Collections.emptyList()))
+              // BigQuery dataset
+              .put(
+                  WsmResourceType.BIG_QUERY_DATASET,
+                  ControlledResourceIamRole.READER,
+                  new CustomGcpIamRole(
+                      WsmResourceType.BIG_QUERY_DATASET,
+                      ControlledResourceIamRole.READER,
+                      BIG_QUERY_DATASET_READER_PERMISSIONS))
+              .put(
+                  WsmResourceType.BIG_QUERY_DATASET,
+                  ControlledResourceIamRole.WRITER,
+                  new CustomGcpIamRole(
+                      WsmResourceType.BIG_QUERY_DATASET,
+                      ControlledResourceIamRole.WRITER,
+                      BIG_QUERY_DATASET_WRITER_PERMISSIONS))
+              .put(
+                  WsmResourceType.BIG_QUERY_DATASET,
+                  ControlledResourceIamRole.EDITOR,
+                  new CustomGcpIamRole(
+                      WsmResourceType.BIG_QUERY_DATASET,
+                      ControlledResourceIamRole.EDITOR,
+                      BIG_QUERY_DATASET_EDITOR_PERMISSIONS))
+              .put(
+                  WsmResourceType.BIG_QUERY_DATASET,
+                  ControlledResourceIamRole.ASSIGNER,
+                  new CustomGcpIamRole(
+                      WsmResourceType.BIG_QUERY_DATASET,
+                      ControlledResourceIamRole.ASSIGNER,
+                      Collections.emptyList()))
```

Comment:
You could use the same convention as the multimaps and omit this entry from the table.

Reply:
I think tables and multimaps treat this differently. In a multimap, a key with no value is the same as the key being there. In a table, having the empty list here is different than not having it